rdflib/term.py

```diff
@@ -1967,14 +1990,21 @@ def _castPythonToLiteral(  # noqa: N802
     (Duration, (lambda i: duration_isoformat(i), _XSD_DURATION)),
     (timedelta, (lambda i: duration_isoformat(i), _XSD_DAYTIMEDURATION)),
     (xml.dom.minidom.Document, (_writeXML, _RDF_XMLLITERAL)),
-    # this is a bit dirty - by accident the html5lib parser produces
-    # DocumentFragments, and the xml parser Documents, letting this
-    # decide what datatype to use makes roundtripping easier, but it a
-    # bit random
-    (xml.dom.minidom.DocumentFragment, (_writeXML, _RDF_HTMLLITERAL)),
     (Fraction, (None, _OWL_RATIONAL)),
 ]
 
+if html5lib is not None:
+    # This is a bit dirty, by accident the html5lib parser produces
+    # DocumentFragments, and the xml parser Documents, letting this
+    # decide what datatype to use makes roundtripping easier, but it a
+    # bit random.
```

Comment:
And this first part of the comment seems to suggest, that DocumentFragments isnt the expected type, although it is. But i have to little knowledge about the roundtripping tests to understand what the author wanted to say otherwise. Maybe we could just drop it.

Reply:
The comment is a bit weird, but what it tries to point out is that it is more or less just luck that gives it a different type from `rdf:XML`, it could be that someone constructs a literal with a `DocumentFragment` which is not representative of HTML, in which case it should not become `rdf:HTML`. This points to some drawbacks in how python values are used to construct literals.

Reply:
mh ok makes sense. You dont know by chance why rdf:xml generates a [document](https://rdflib.readthedocs.io/en/stable/rdf_terms.html#python-conversions) instead of a [documentfragment](https://www.w3.org/TR/rdf12-concepts/#section-XMLLiteral)?